Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.hxx

```diff
@@ -101,7 +101,16 @@ namespace AzToolsFramework
 
         void SetComponentOverridden(const bool overridden);
 
-        void EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes);
+        enum ComponentEditorState
+        {
+            DidEnterComponentMode,
+            DidNotEnterComponentMode
+        };
```

Comment:
update: Thanks for adding the enum @amzn-ahmadkrm but if you see my earlier comment I think we can have more accurate names - also we should use `enum class` here not just `enum`